File: fractions.java
import java.util.Arrays;
import java.util.Scanner;

/*
Solution to Fractions by David Harmeyer

Observation: assume you have three numbers a, b, and c, where 0 < a < b < c.
If c is divisible by b and b is divisible by a, then c is divisible by a.
The proof of this left to the reader, but pretty intuitive if you think about it.

Personally, I felt that this problem would have been way more interesting if we set n to
10^5, because with n=1000 you can literally just check all n*(n-1)/2 pairs.

 */
public class fractions {

	public static void main(String[] args) {
		Scanner fs=new Scanner(System.in);
		//read in input
		int T=fs.nextInt();
		for (int t=1; t<=T; t++) {
			int n=fs.nextInt();
			int[] a=new int[n];
			for (int i=0; i<n; i++)
				a[i]=fs.nextInt();
			
			//make sure each pair of adjacent elements in the sorted array works
			Arrays.sort(a);
			boolean works=true;
			for (int i=1; i<n; i++)
				works&=a[i]%a[i-1]==0;
			
			System.out.println("Array #"+t+": "+(works?"This array is bae!":"Go away!"));
		}
	}

}

File: fractions.py
#John Edwards
#2019-02-13

# O(n**2) version
# handle input and sort array
for t in range(int(input())):
    n = int(input())
    l = list(map(int, input().split()))
    # loop over all pairs and count the number
    # of pairs that are good
    cnt = 0
    for i in range(n):
        for j in range(n):
            if l[i] %l[j] == 0 or l[j] %l[i] == 0:
                cnt += 1
    # if all pairs are good, this array is bae
    if cnt == n**2:
        print("Array #"+str(t+1)+": This array is bae!")
    else:
        print("Array #"+str(t+1)+": Go away!") 